### PR TITLE
Adding CDH version of the sparklyr demo

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: dev/flights-cdh/sparkClusterDemo-source.R [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+library(sparklyr)
+library(dplyr)
+library(ggplot2)
+
+Sys.setenv(HADOOP_CONF_DIR='/etc/hadoop/conf.cloudera.hdfs')
+Sys.setenv(YARN_CONF_DIR='/etc/hadoop/conf.cloudera.yarn')
+Sys.setenv(SPARK_HOME="/home/ubuntu/spark-1.6.0")
+Sys.setenv(SPARK_HOME_VERSION="1.6.0")
+
+sc <- spark_connect(master = "yarn-client", version="1.6.0", spark_home = '/opt/cloudera/parcels/CDH/lib/spark/')
```
Comment (slopp) [inline dev/flights-cdh/sparkClusterDemo-source.R:11]: doesn't this line over-ride the previous lines?

**Comment by slopp** [comment]
It seems very strange that we have to install an entirely new distribution of spark, only to over-ride the definition of SPARK_HOME with:

``` r
sc <- spark_connect(master = "yarn-client", version="1.6.0", spark_home = '/opt/cloudera/parcels/CDH/lib/spark/')
```

Is it really necessary to install java, scala, and spark only to use the Cloudera bundle?  Perhaps @javierluraschi  can weigh in since he is work on the Cloudera cert?

If it is necessary to install a second version of Spark can we simplify the process using sparklyr's installer? ie
`Rscript -e "sparklyr::spark_install(...)"`

If that doesn't work, then we should at least remove the dependencies on vi; we can directly modify the .bashrc file:

`echo "SPARK_HOME=...." >>  .bashrc`

File: dev/flights-cdh/sparkClusterDemo-source.R (hunk 1 above)
Comment (ghost) [reply dev/flights-cdh/sparkClusterDemo-source.R:11]: That's what I thought too, but if I leave out line 8, then the sparklyr job fails saying that it cannot find certain files in the Spark parcel folder

**Comment by javierluraschi** [comment]
@slopp @edgar-rstudio this is how I would connect to `Cloudera`:

``` r
Sys.setenv(SPARK_HOME = "/opt/cloudera/parcels/CDH/lib/spark/")
sc <- spark_connect(master = "yarn-client", version = "1.6.0")
```

Correct, we don't need a full installation with `spark_install`; therefore, this can be removed: `Sys.setenv(SPARK_HOME="/home/ubuntu/spark-1.6.0")` or replaced with the CDF `SPARK_HOME` path as I mention above.

There should only be one `SPARK_HOME` or `spark_home` parameter defined, since they both set the same context.

The `version` parameter and the `SPARK_HOME_VERSION` are also intended to be equivalent so we should only define one. However, the CRAN version of `sparklyr` contains a bug where this is not respected, I believe this should be fixed with this in newer builds: https://github.com/rstudio/sparklyr/pull/268 - Therefore, using the `version` parameter is preferred for our examples.

**Pull request merged**